src/test/HostActivationTests/GivenThatICareAboutStandaloneAppActivation.cs

```diff
@@ -150,6 +151,77 @@ public void Running_Publish_Output_Standalone_EXE_By_Renaming_dotnet_exe_Fails()
             }
         }
 
+        [Fact]
+        public void Running_Publish_Output_Standalone_EXE_By_Renaming_apphost_exe_Succeeds()
+        {
+            var fixture = PreviouslyPublishedAndRestoredStandaloneTestProjectFixture
+                .Copy();
+
+            var appExe = fixture.TestProject.AppExe;
+            var renamedAppExe = fixture.TestProject.AppExe + $"renamed{Constants.ExeSuffix}";
+
+            File.Copy(appExe, renamedAppExe, true);
+
+            // TODO: Use FS.Chmod when build utility project is converted to csproj.
+            // See https://github.com/NuGet/Home/issues/4424
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Command.Create("chmod", "u+x", renamedAppExe).Execute().EnsureSuccessful();
+            }
+
+            Command.Create(renamedAppExe)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hello World");
+        }
+
+        [Fact]
+        public void Running_Publish_Output_Standalone_EXE_With_Startupconfig_Succeeds()
```

Comment:
Do you also want a test that sets the environment variable?

Reply:
I'll add one

Reply:
I'll add a basic test for the env var.

Note the env var is only used for apphost + framework-dependent I've only done manual testing for that because there is no tooling for that yet and the workarounds at this point are ugly (publish for both self-contained and framework-dependent). 

So I will add a test adds a simple check to make sure a self-contained apphost tries to use DOTNET_ROOT, but the actual app launch fails due to other (file layout) reasons.